### ISSUE TITLE
FindSortableProperties - The type '{type.FullName}' has not been declared in the entity data model.

### ISSUE THREAD

**Issue by fmclopes** [AutoMapper/AutoMapper.Extensions.OData#194]
Hello,

I have find a problem when using the method GetQueryAsync. When I use this method give the error "The type 'Module1.CountryDto' has not been declared in the entity data model."

I have a EDM Model with 2 Dto with the same name but in different modules:
- Module1.CountryDto
- Module2.CountryDto

The method FindSortableProperties have a function called GetEntity the find by name, but I think it should be by full name. In my case the entities.Count will give 2 and will return null.

```C#
IEdmEntityType GetEntity()
{
    List<IEdmEntityType> entities = context.Model.SchemaElements.OfType<IEdmEntityType>().Where(e => e.Name == type.Name).ToList();
    if (entities.Count == 1)
        return entities[0];

    return null;
}
```

Note: when i use $orderBy it work because not call the method FindSortableProperties.

### Version

AutoMapper.AspNetCore.OData.EFCore 4.0.0

### Expected behavior

No error.

### Actual behavior

Request results in status code 500 Internal Server Error with argument exception:

```
The type 'Module1.Country' has not been declared in the entity data model.
at AutoMapper.AspNet.OData.ODataQueryContextExtentions.FindSortableProperties(ODataQueryContext context, Type type)
   at AutoMapper.AspNet.OData.LinqExtensions.GetQueryableMethod(Expression expression, ODataQueryContext context, OrderByClause orderByClause, Type type, Nullable`1 skip, Nullable`1 top)
   at AutoMapper.AspNet.OData.LinqExtensions.GetOrderByMethod[T](Expression expression, ODataQueryOptions`1 options, ODataSettings oDataSettings)
   at AutoMapper.AspNet.OData.LinqExtensions.GetQueryableExpression[T](ODataQueryOptions`1 options, ODataSettings oDataSettings)
   at AutoMapper.AspNet.OData.QueryableExtensions.GetQueryable[TModel,TData](IQueryable`1 query, IMapper mapper, ODataQueryOptions`1 options, QuerySettings querySettings, Expression`1 filter)
   at AutoMapper.AspNet.OData.QueryableExtensions.<GetQueryAsync>d__2`2.MoveNext()
```

**Comment by BlaiseD** [comment]
The comparison by name fixes issues #149  and #152.  We could still check for FullName if there's more than result. Maybe: 
```c#
            IEdmEntityType GetEntity()
            {
                List<IEdmEntityType> entities = context.Model.SchemaElements.OfType<IEdmEntityType>().Where(e => e.Name == type.Name).ToList();
                if (entities.Count == 1)
                    return entities[0];

                if (entities.Count > 1)
                    return entities.FirstOrDefault(e => e.FullName() == type.FullName);

                return null;
            }
```
or whatever works for the full name comparison.

PRs are welcome.

**Comment by BlaiseD** [comment]
I think you may have to assign a [custom namespace](https://github.com/AutoMapper/AutoMapper.Extensions.OData/blob/675de0628bf24372ed7182549c587e9e3d460f80/AutoMapper.OData.EFCore.Tests/GetTests.cs#L686) to the builder.  Here's a [test doing that](https://github.com/AutoMapper/AutoMapper.Extensions.OData/blob/675de0628bf24372ed7182549c587e9e3d460f80/AutoMapper.OData.EFCore.Tests/GetTests.cs#L262).  Issues #149 and #152 might also shed some light.

**Comment by maboivin** [comment]
I also ran into this issue on my side but the cause is different.

In my case, when we build the OData model, we change the name of the entity types by removing the suffix `Model`. We also use a custom namespace. 

For example, `MyEntityModel` becomes `MyEntity` and the namespace 'My.App' becomes 'My.ODataApi'. Obviously, in my case, the entity is just not found (`entities.Count = 0`). 

ASP.NET Core OData seems to be using `IODataTypeMapperExtensions` and `IODataTypeMapper`. Would it make sense to use it here as well to get the right mapping between the OData entity type and the CLR type?

**Comment by maboivin** [comment]
@BlaiseD It looks like the fix proposed in #226 doesn't solve my issue where the name of the EDM type in the OData model is different than the name of the CLR type.

In my case, the query to get the list of EDM entity types by name doesn't return any entry.

Should I open another issue? Would you like a simple repro project?

**Comment by BlaiseD** [comment]
You can create a new issue or a PR to fix it.  The solution is probably to use the [CLR filter here](https://github.com/AutoMapper/AutoMapper.Extensions.OData/blob/6a68bac6af428e34bb3e60fb8d43f4fcd1ead469/AutoMapper.AspNetCore.OData.EFCore/ODataQueryContextExtentions.cs#L24) and that should cover all cases.

You'll need a failing test to cover your use case.
